Code/Engine/GameEngine/CMakeLists.txt

```diff
@@ -9,17 +9,21 @@ ez_create_target(LIBRARY ${PROJECT_NAME})
 
 target_link_libraries(${PROJECT_NAME}
   PRIVATE
-  RendererDX11
 )
 
 target_link_libraries(${PROJECT_NAME}
   PUBLIC
   RendererCore
   ozz
+)
+
+add_dependencies(${PROJECT_NAME}
+  RendererDX11
+  RendererVulkan
```

Comment:
@jankrassnigg Any idea how to add this dependency without getting an error in cmake if RendererVulkan does not exist (when EZ_BUILD_EXPERIMENTAL_VULKAN is not enabled)

Reply:
I think so. Will have a look later.

Reply:
In CMake you can check whether a target exists:

  if (TARGET blub)
    add_dependencies(bla blub)
  endif()

So that's how you can add the dependency only if CMake knows about it. Unfortunately, CMake evaluates that if right away, and therefore may not know about a target, that will be declared later. That's why I introduced the 'finalize.cmake' files. Those are run after all the projects are declared. So if you put such a file into this directory, you can do that check and it should always work.

Just have a look at some other finalize.cmake file, they are all used for similar purposes.

Reply:
I've added the following to finalize.cmake for the GameEngine:

if (TARGET RendererDX11)
  add_dependencies(${PROJECT_NAME} RendererDX11)
endif()

then cmake complains:

CMake Error at Code/Engine/GameEngine/finalize.cmake:2 (add_dependencies):
  Cannot add target-level dependencies to non-existent target "ezVs2019".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.
Call Stack (most recent call first):
  Code/BuildSystem/CMake/ezUtils.cmake:477 (include)
  CMakeLists.txt:65 (ez_finalize_projects)